waku/incentivization/rpc.nim

```diff
@@ -0,0 +1,27 @@
+import
+  json_serialization,
+  std/options
+import
+  ../waku_core
+
+type
+  EligibilityProofType* {.pure.} = enum
+    # Indicates the type of eligibility proof
+    NONE = uint32(0)
+    TX_ID = uint32(1)
+
+  EligibilityProof* = object
+    proofType*: EligibilityProofType
+    proof*: string # or bytes?
+
+  DummyRequest* = object
+    requestId*: string
+    eligibilityProof*: Option[EligibilityProof]
```

Comment:
I think both `eligibilityProof` and `eligibilityStatus` should be `seq[bytes]` within their encapsulated Request and Response messages respectively (and in the protobufs `bytes`). This allows for two stage decoding (those decoding `DummyRequest` don't need to also decode `EligibilityProof` if not interesting to them). I see the example encoding into HistoryRequest and HistoryResponse in https://github.com/vacp2p/rfc/tree/master/content/docs/rfcs/73 is inconsistent here - encoding as `bytes` in the Request but not in the Response.

Reply:
> both eligibilityProof and eligibilityStatus should be seq[bytes] within their encapsulated Request and Response messages respectively (and in the protobufs bytes)

I find this a bit confusing: how does this suggestion relate to the RFC definitions of `eligibilityProof` and `eligibilityProof`? For example, `eligibilityStatus` is defined in the RFC as an object containing an optional `uint32` and an optional `string`. How does this relate to treating both `eligibilityProof` and `eligibilityProof` as bytes? Do you suggest we change the RFC in that respect?

> the example encoding into HistoryRequest and HistoryResponse in https://github.com/vacp2p/rfc/tree/master/content/docs/rfcs/73 is inconsistent here - encoding as bytes in the Request but not in the Response.

Do you mean, the exampes in the RFC are inconsistent? AFAIR, my reasoning was that the proof can take many forms (txid, zkp, etc), which we don't know in advance and therefore use `bytes` as the lowest common denominator, but the response, in contrast, always has the same form: an error code and an error description.

Reply:
The reasoning makes perfect sense, thank you! So am I correct that this PR is about the eligibility module that "decodes/encodes/interprets eligibility protocol"? Which means that the codec defined in this PR _should_ understand the inner structure of `EligibilityStatus`, namely, that it contains a status code and status description? (Sorry if this looks confusing in an "outdated" thread.)

Reply:
Indeed! This comment is outdated in that we don't have the Dummy Protocol anymore, but will be relevant for how this gets encapsulated in future req-resp protocols.